pkg/kubelet/metrics/metrics.go

```diff
@@ -483,6 +483,39 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		},
 	)
+
+	// ImagePullDuration is a Histogram that tracks the duration (in seconds) it takes to pull Images.
+	ImagePullDuration = metrics.NewHistogramVec(
```

Comment:
are these metrics already covered by the runtime operation metrics (duration, operations, errors?)

Reply:
Not exactly, it doesn't have additional labels such as image pull handler, pull state, image, i think these make sense.

Reply:
Similar to metrics RunPodSandboxDuration, RunPodSandboxErrors.

Reply:
I cannot find containerd metrics for image pulling. 
I prefer to add an image pulling in container runtime. (I opened https://github.com/containerd/containerd/issues/7241 to track it in containerd)

Reply:
> I cannot find containerd metrics for image pulling. I prefer to add an image pulling in container runtime. (I opened [containerd/containerd#7241](https://github.com/containerd/containerd/issues/7241) to track it in containerd)

According to CRI, the container runtime implementation may change conditionally such as cri-o/cri-containerd/docker/rkt/frakti, kubelet image manager is the best place to to this work.

Reply:
> it doesn't have additional labels such as image pull handler, pull state, image, i think these make sense.

Image in particular would  have too high of cardinality.

Reply:
I wonder if the grpc code could be added to the runtime operations metrics to solve state.

Reply:
> 

Got it.

Reply:
> > it doesn't have additional labels such as image pull handler, pull state, image, i think these make sense.
> 
> Image in particular would have too high of cardinality.

If so,  the runtime operation metrics maybe the best choice.

Reply:
> I wonder if the grpc code could be added to the runtime operations metrics to solve state.

For image pulling count observation, I have made some changes in image_manager.go. 
PTAL, thanks.